### PR TITLE
fix(legacy proto load): use single proto file

### PR DESCRIPTION
**Pull request by alexander-fenster**
Also, removes proto JSON file if legacy proto options is enabled since it's not used but consumes space and takes time to generate.

### CODE REVIEW

**Comment by alexander-fenster** [comment]
Cc: @aohren

**Comment by aohren** [comment]
Tested and it looks like this works. And there are some noticeable performance improvements.

**Review by summer-ji-eng — APPROVED** [review]

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 1]
```diff
@@ -253,11 +247,9 @@ export class {{ service.name }}Client {
 {%- if api.legacyProtoLoad %}
 
     let protoFilesRoot = new this._gaxModule.GoogleProtoFilesRoot();
-{%- for filename in api.filesToGenerate %}
```
Comment (summer-ji-eng) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk:256]: One question, why do we need to remove the filesToGenerate condition? We only has one proto to generate?

Comment (alexander-fenster) [reply templates/typescript_gapic/src/$version/$service_client.ts.njk:256]: Yes, the idea is to have just one proto here (the one that defines the service).  If we find it does not work for some use cases, we'll think how to fix it.

**Pull request merged**